aare/aare.py

```diff
@@ -1,15 +1,66 @@
 import click
 import requests
+import itertools
+import datetime
 
 
 @click.command()
-def aare():
+@click.option('--statistics', help='show the hottest and the coolest temperature of the last 7 days', is_flag="True")
+def aare(statistics):
     """A simple command which displays the current aare temperature"""
+    if statistics:
+        display_stats()
+    else:
+        display_current_temp()
+
+
+def display_current_temp():
     api_url = 'http://aare.schwumm.ch/aare.json'
     response = requests.get(api_url)
     aare_json = response.json()
     click.echo("Current temperature of the aare: {}° C".format(aare_json["temperature"]))
 
 
+def get_min_max_date_of_datetime_temp_group(group):
+    min_temp = 0
+    max_temp = 0
+    day = ""
+    initial = True
+    # group contains no index, therefore inital-bool is used to get the first entry
+    for item in group:
+        if initial:
+            min_temp = item[1]
+            max_temp = item[1]
+            day = item[0].strftime("%Y-%m-%d")
+            initial = False
+        if min_temp > item[1]:
+            min_temp = item[1]
+        if max_temp < item[1]:
+            max_temp = item[1]
+    return min_temp, max_temp, day
+
+
+def display_stats():
+    api_url = 'http://aare.schwumm.ch/api/archive'
+    params = {"from": "now", "to": "-7 days"}
+    # The website denies unknown user agents. So we have to impersonate a general webbrowser to gain access
+    headers = {
+        'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) '
+                      'Chrome/50.0.2661.102 Safari/537.36'}
+    response = requests.get(api_url, params, headers=headers)
+    response_json = response.json()
+    temperatures = response_json['data']['temperature']
+    dates = [datetime.datetime.strptime(item, '%Y-%m-%d %H:%M:%S') for item in response_json['data']['datetime']]
+    data = list(zip(dates, temperatures))
+
+    # Remove data points not having a temperature
+    data = list(filter(lambda item: item[1], data))
+    for key, group in itertools.groupby(data, lambda item: datetime.date(item[0].year, item[0].month, item[0].day)):
+        # group can be iterated only once because after the iteration it will be cleared automatically
+        min_temp, max_temp, day_string = get_min_max_date_of_datetime_temp_group(group)
```

Comment:
You could use this instead:

```python
temperatures = [item[1] for item in tuple(group)]
min_temp = min(temperatures)
max_temp = max(temperatures)
day_string = key.strftime('%Y-%m-%d')
```

Reply:
I thought it would be a little more performant if I define and use the `get_min_max_date_of_datetime_temp_group`-method, since the method will iterate only once while your implementation will iterate at least 3 times (depending of the implementation of `min` and `max`)

But your approach seems more readable and performance is not an issue, I will change the corresponding lines.